controllers/datamovement_controller.go

```diff
@@ -214,40 +254,154 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
+		// last regex match to obtain the most recent progress.
+		go func() {
+			var progressInt32 *int32
+			var elapsed metav1.Duration
+			elapsed.Duration = 0
+			progressStart := metav1.NowMicro()
+
+			// Perform the actual collection and update logic
+			parseAndUpdateProgress := func() {
+				progressOutput := ""
+
+				// Read all lines of output until EOF
+				for {
+					line, err := parseBuf.ReadString('\n')
+					if err == io.EOF {
+						break
+					} else if err != nil {
+						log.Error(err, "failed to read progress output")
+					}
+					progressOutput += line
+				}
+
+				// Get the times now before we start the parsing
+				progressNow := metav1.NowMicro()
+				elapsed.Duration = progressNow.Time.Sub(progressStart.Time)
+
+				// Find all matching progress matches but we only care about the
+				// last (most recent one)
+				groups := progressRe.FindAllStringSubmatch(progressOutput, -1)
+				if len(groups) > 0 {
+					mostRecent := groups[len(groups)-1]
+
+					// Get the progress and convert to int
+					progress, err := strconv.Atoi(mostRecent[3])
+					if err == nil {
+						// Initialize the pointer if we haven't done so already.
+						// We don't want to do this above since we do not want an
+						// initial value of 0 unless we actually have a progress
+						if progressInt32 == nil {
+							progressInt32 = new(int32)
+						}
+						*progressInt32 = int32(progress)
+					} else {
+						log.Error(err, "failed to parse progress output", "progressMatch", mostRecent)
+					}
+				}
+
+				// Update the CommandStatus in the DM resource
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					dm := &nnfv1alpha1.NnfDataMovement{}
+					if err := r.Get(ctx, req.NamespacedName, dm); err != nil {
+						return client.IgnoreNotFound(err)
+					}
+
+					// Update progress if we have one
+					if progressInt32 != nil {
+						cmdStatus.ProgressPercentage = progressInt32
+					}
+					// Update the output and times regardless
+					cmdStatus.LastMessage = progressOutput
+					cmdStatus.LastMessageTime = progressNow
+					cmdStatus.ElapsedTime = elapsed
+
+					if dm.Status.CommandStatus == nil {
+						dm.Status.CommandStatus = &nnfv1alpha1.NnfDataMovementCommandStatus{}
+					}
+					cmdStatus.DeepCopyInto(dm.Status.CommandStatus)
+
+					log.Info("Updating Progress", "CommandStatus", dm.Status.CommandStatus)
+					return r.Status().Update(ctx, dm)
+				})
+
+				if err != nil {
+					log.Error(err, "failed to update CommandStatus with Progress", "CommandStatus", cmdStatus)
+				}
+			}
 
-		// TODO: Capture output as it progresses and add a %complete to the resource
+			// Main Progress Collection Loop
+			for {
+				select {
+				// Now that we're done, parse whatever output is left
+				case <-chCommandDone:
+					parseAndUpdateProgress()
+					chProgressDone <- true
+					return
+				// Collect Progress output on every interval
+				case <-time.After(progressCollectInterval):
```

Comment:
Should we allow for progressCollectInterval to be 0? What about negative (turn off collection)?

Reply:
I think if the interval is <1s, we should disable progress collection. I'll add that.